README.md

```diff
@@ -178,6 +178,87 @@ impl Counter {
 To upgrade the contract first call `up_stage_code` passing the binary as first argument serialized as borsh. Then call `up_deploy_code`.
 This functions must be called from the owner.
 
+### [AccessControllable](/near-plugins/src/access_controllable.rs)
+
+Enables role based access control for contract methods. A method with restricted access can only be called _successfully_ by accounts that have been granted one of the whitelisted roles. If a restricted method is called by an account with insufficient permissions it panics.
+
+Each role is managed by admins who may grant the role to accounts and revoke it from them. In addition, there are super admins that have admin permissions for every role.
+
+The sets of accounts that are (super) admins and grantees are stored in the contract's state.
+
+```rust
+/// Roles are represented by enum variants.
+/// 
+/// Deriving `AccessControlRole` ensures `Role` can be used in
+/// `AccessControllable`.
+#[derive(AccessControlRole, Deserialize, Serialize, Copy, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub enum Role {
+    SkipperByOne,
+    SkipperByAny,
+    Resetter,
+}
+
+#[access_control(role_type(Role))]
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+pub struct Contract {
+    counter: u64,
+}
+
+#[near_bindgen]
+impl Counter {
+    /// Setup access control in the constructor.
+    #[init]
+    pub fn new() -> Self {
+        let contract = Self {
+            counter: 0,
+            __acl: Default::default(), // initialize state
+        };
+
+        // Add a super admin.
+        near_sdk::require!(
+            contract.acl_init_super_admin(near_sdk::env::predecessor_account_id()),
+            "Failed to initialize super admin",
+        );
+
+        // Add an admin. This is possible since the contract was just made a
+        // super admin.
+        near_sdk::require!(
+            Some(true) == contract.acl_add_admin(Role::SkipperByOne.into(), admin_account_id),
+            "Failed to add admin",
+        );
+
+        // Grant a role. Also possible since the contract is super admin.
+        near_sdk::require!(
+            Some(true) == contract.acl_grant_role(Role::SkpperByAny.into(), grantee_account_id),
+            "Failed to grant role",
+        );
```

Comment:
I think it's worth adding variables for `admin_account_id` and `grantee_account_id` just to show that these are user-defined.

E.g.:
```rust
// Define the account you would like to be an admin
let skipper_by_one_admin_account_id: AccountId = "alice.near".parse().unwrap();
```

It's also worth adding a comment for the "add an admin" part that we are adding an admin just for the specific role. Even though it's mentioned before that admins and super-admins are different in the ACL hierarchy, I don't think the comment would be redundant.